crates/symbolicator/src/services/symbolication/process_minidump.rs

```diff
@@ -0,0 +1,1227 @@
+                        CacheStatus::CacheSpecificError(details) => {
+                            let err = CfiCacheError::ObjectParsing(ObjectError::Malformed);
+                            tracing::warn!("Cached error from parsing cficache: {}", details);
+                            ObjectFileStatus::from(&err)
+                        }
+                    };
+                    let cfi_path = match cfi_cache.status() {
+                        CacheStatus::Positive => Some(cfi_cache.path().to_owned()),
+                        _ => None,
+                    };
+                    CfiModule {
+                        features: cfi_cache.features(),
+                        cfi_status,
+                        cfi_path,
+                        cfi_candidates: cfi_cache.candidates().clone(), // TODO(flub): fix clone
+                        ..Default::default()
+                    }
+                }
+                Err(err) => {
+                    let stderr: &dyn std::error::Error = &err;
+                    tracing::debug!(stderr, "Error while fetching cficache");
+                    CfiModule {
+                        cfi_status: ObjectFileStatus::from(err.as_ref()),
+                        ..Default::default()
+                    }
+                }
+            };
+            (code_id, cfi_module)
+        });
+        self.inner.extend(iter)
+    }
+
+    /// Returns a mapping of module IDs to paths that can then be loaded inside a procspawn closure.
+    fn for_processing(&self) -> Vec<(DebugId, PathBuf)> {
+        self.inner
+            .iter()
+            .filter_map(|(id, module)| Some((*id, module.cfi_path.clone()?)))
+            .collect()
+    }
+
+    /// Returns the inner Map.
+    fn into_inner(self) -> BTreeMap<DebugId, CfiModule> {
+        self.inner
+    }
+}
+
+fn object_info_from_minidump_module(ty: ObjectType, module: &MinidumpModule) -> RawObjectInfo {
+    // Some modules are not objects but rather fonts or JIT areas or other mmapped files
+    // which we don't care about.  These may not have complete information so map these to
+    // our schema by converting to None when needed.
+    let code_id = module
+        .code_identifier()
+        .filter(|code_id| !code_id.is_nil())
+        .map(|code_id| code_id.to_string().to_lowercase());
+    let code_file = module.code_file();
+    let code_file = match code_file.is_empty() {
+        true => None,
+        false => Some(code_file.into_owned()),
+    };
+
+    RawObjectInfo {
+        ty,
+        code_id,
+        code_file,
+        debug_id: module.debug_identifier().map(|c| c.breakpad().to_string()),
+        debug_file: module.debug_file().map(|c| c.into_owned()),
+        image_addr: HexValue(module.base_address()),
+        image_size: match module.size() {
+            0 => None,
+            size => Some(size),
+        },
+    }
+}
+
+async fn stackwalk_with_rust_minidump(
```

Comment:
did you leave this name intentionally?  you mentioned having removed these.  i'd be up for simply calling this `stackwalk`